Persephone/Controllers/QueueViewController.swift

```diff
@@ -8,210 +8,150 @@
     guard let state = notification.userInfo?[Notification.stateKey] as? MPDClient.Status.State
       else { return }
 
-    setQueueIcon(state)
+    dataSource.setQueueIcon(state)
   }
 
   @objc func queueChanged(_ notification: Notification) {
     guard let queue = notification.userInfo?[Notification.queueKey] as? [MPDClient.Song]
       else { return }
 
-    self.queue = queue
-
+    dataSource.updateQueue(queue)
     queueView.reloadData()
   }
 
   @objc func queuePosChanged(_ notification: Notification) {
     guard let queuePos = notification.userInfo?[Notification.queuePosKey] as? Int
       else { return }
 
-    let oldSongRowPos = self.queuePos + 1
-    let newSongRowPos = queuePos + 1
-    self.queuePos = queuePos
-
-    setQueuePos(oldSongRowPos: oldSongRowPos, newSongRowPos: newSongRowPos)
-
-    queueView.reloadData(
-      forRowIndexes: [oldSongRowPos, newSongRowPos],
-      columnIndexes: [0, 1]
-    )
-  }
-
-  @objc func clearQueue(_ notification: Notification) {
-    self.queue = []
+    dataSource.setQueuePos(queuePos)
 
     queueView.reloadData()
   }
 
-  func setQueueIcon(_ state: MPDClient.Status.State) {
-    switch state {
-    case .playing:
-      self.queueIcon = playIcon
-    case .paused:
-      self.queueIcon = pauseIcon
-    default:
-      self.queueIcon = nil
-    }
-  }
-
-  func setQueuePos(oldSongRowPos: Int, newSongRowPos: Int) {
-    if oldSongRowPos > 0 {
-      guard let oldSongRow = queueView.rowView(atRow: oldSongRowPos, makeIfNecessary: true),
-        let oldSongTitleCell = oldSongRow.view(atColumn: 0) as? NSTableCellView
-        else { return }
-
-      setRowFont(rowView: oldSongRow, font: systemFontRegular)
-      oldSongTitleCell.imageView?.image = nil
-    }
-
-    guard let songRow = queueView.rowView(atRow: newSongRowPos, makeIfNecessary: true),
-      let newSongTitleCell = songRow.view(atColumn: 0) as? NSTableCellView
-      else { return }
-
-    setRowFont(rowView: songRow, font: systemFontBold)
-    newSongTitleCell.imageView?.image = self.queueIcon
-  }
-
-  func setRowFont(rowView: NSTableRowView, font: NSFont) {
-    guard let songTitleCell = rowView.view(atColumn: 0) as? NSTableCellView,
-      let songArtistCell = rowView.view(atColumn: 1) as? NSTableCellView
-      else { return }
-
-    songTitleCell.textField?.font = font
-    songArtistCell.textField?.font = font
-  }
-
-  func outlineView(_ outlineView: NSOutlineView, numberOfChildrenOfItem item: Any?) -> Int {
-    return queue.count + 1
-  }
-
-  func outlineView(_ outlineView: NSOutlineView, isItemExpandable item: Any) -> Bool {
-    return false
-  }
-
-  func outlineView(_ outlineView: NSOutlineView, child index: Int, ofItem item: Any?) -> Any {
-    if index > 0 {
-      return SongItem(song: queue[index - 1], queuePos: index - 1)
+  func outlineView(
+    _ outlineView: NSOutlineView,
+    selectionIndexesForProposedSelection proposedSelectionIndexes: IndexSet
+    ) -> IndexSet {
```

Comment:
swiftformat?

Reply:
I'm pretty close to just reformatting all my guards so they work with swiftformat!